components/observations/ObservationsFilterForm.tsx

```diff
@@ -123,26 +145,54 @@ const matchesAvalancheInstability = (instability: avalancheInstability, observat
   throw new Error(`Unknown instability: ${invalid}`);
 };
 
-export const filtersForConfig = (mapLayer: MapLayer, config: ObservationFilterConfig, currentDate: Date): FilterFunction[] => {
+interface FilterListItem {
+  filter: FilterFunction;
+  label: string;
+  removeFilter?: (config: ObservationFilterConfig) => ObservationFilterConfig;
+}
+export const filtersForConfig = (mapLayer: MapLayer, config: ObservationFilterConfig, currentDate: Date): FilterListItem[] => {
   if (!config) {
     return [];
   }
 
-  const filterFuncs: FilterFunction[] = [];
-  if (config.zone) {
-    filterFuncs.push(observation => config.zone === matchesZone(mapLayer, observation.locationPoint?.lat, observation.locationPoint?.lng));
+  const filterFuncs: FilterListItem[] = [];
+  if (config.dates && config.dates.value) {
+    filterFuncs.push({filter: matchesDates(currentDate, config.dates), label: dateLabelForFilterConfig(config.dates)});
   }
 
-  if (config.dates && config.dates.value) {
-    filterFuncs.push(matchesDates(currentDate, config.dates));
+  if (config.zone) {
+    filterFuncs.push({
+      filter: observation => config.zone === matchesZone(mapLayer, observation.locationPoint?.lat, observation.locationPoint?.lng),
+      label: config.zone,
+      removeFilter: config => ({...config, zone: undefined}),
```

Comment:
Is there a reason to choose a spread here (which would allocate a whole new object & copy all data, right?) over just mutating in place?

Reply:
Yeah, it's intentional, the input comes from a call to `useState` (so should be treated as immutable) and the return value needs to be passed to `setFilterConfig`.

Compared to the number of object copies we do in the observation list proper, I guarantee you this is not worth worrying about from a perf perspective 🙈 

But I do take the point that the naming here (`removeFilter`) is super-confusing, because it's not actually removing the filter. Let me know if you have better ideas!

Reply:
Nope, your explanation makes sense, nothing needs to change here